randomwallpaper@iflow.space/ui/wallhaven.js

```diff
@@ -30,53 +35,108 @@ var WallhavenSettingsGroup = GObject.registerClass({
+		this._settings.bind('allow-sketchy',
+			this._allow_sketchy,
+			'active',
 			Gio.SettingsBindFlags.DEFAULT);
 		this._settings.bind('api-key',
 			this._api_key,
 			'text',
 			Gio.SettingsBindFlags.DEFAULT);
-		this._settings.bind('resolutions',
-			this._resolutions,
-			'text',
+		this._settings.bind('category-anime',
+			this._category_anime,
+			'active',
 			Gio.SettingsBindFlags.DEFAULT);
 		this._settings.bind('category-general',
 			this._category_general,
 			'active',
 			Gio.SettingsBindFlags.DEFAULT);
-		this._settings.bind('category-anime',
-			this._category_anime,
-			'active',
-			Gio.SettingsBindFlags.DEFAULT);
 		this._settings.bind('category-people',
 			this._category_people,
 			'active',
 			Gio.SettingsBindFlags.DEFAULT);
-		this._settings.bind('allow-sfw',
-			this._allow_sfw,
-			'active',
+		this._settings.bind('color',
+			this._row_color,
+			'subtitle',
 			Gio.SettingsBindFlags.DEFAULT);
-		this._settings.bind('allow-sketchy',
-			this._allow_sketchy,
-			'active',
+		this._settings.bind('keyword',
+			this._keyword,
+			'text',
 			Gio.SettingsBindFlags.DEFAULT);
-		this._settings.bind('allow-nsfw',
-			this._allow_nsfw,
-			'active',
+		this._settings.bind('resolutions',
+			this._resolutions,
+			'text',
 			Gio.SettingsBindFlags.DEFAULT);
+
+		this._button_color_undo.connect('clicked', () => {
+			this._row_color.subtitle = "";
+		});
+
+		const availableColors = [
+			"#660000", "#990000", "#cc0000", "#cc3333", "#ea4c88",
+			"#993399", "#663399", "#333399", "#0066cc", "#0099cc",
+			"#66cccc", "#77cc33", "#669900", "#336600", "#666600",
+			"#999900", "#cccc33", "#ffff00", "#ffcc33", "#ff9900",
+			"#ff6600", "#cc6633", "#996633", "#663300", "#000000",
+			"#999999", "#cccccc", "#ffffff", "#424153",
+		];
+
+		this._colorPalette = [];
+
+		availableColors.forEach(hexColor => {
+			let rgbaColor = new Gdk.RGBA();
+			rgbaColor.parse(hexColor);
+			this._colorPalette.push(rgbaColor);
+		});
+
+		this._button_color.connect('clicked', () => {
+			// For GTK 4.10+
+			// Gtk.ColorDialog();
+
+			// https://stackoverflow.com/a/54487948
+			this._colorDialog = new Gtk.ColorChooserDialog({
+				title: 'Choose a Color',
+				transient_for: this.get_root(),
+				modal: true,
+			});
+			this._colorDialog.set_use_alpha(false);
+			this._colorDialog.add_palette(Gtk.Orientation.HORIZONTAL, 10, this._colorPalette);
+
+			this._colorDialog.connect('response', (dialog, response_id) => {
+				if (response_id === Gtk.ResponseType.OK) {
+					// result is a Gdk.RGBA which uses float
+					let rgba = this._colorDialog.get_rgba();
+					// convert to rgba so it's useful
+					let rgbaString = rgba.to_string(); // rgb(0,0,0)
+					let rgbaArray = rgbaString.replace("rgb(", "").replace(")", "").split(",")
+					let hexString = `${parseInt(rgbaArray[0]).toString(16).padStart(2, "0")}${parseInt(rgbaArray[1]).toString(16).padStart(2, "0")}${parseInt(rgbaArray[2]).toString(16).padStart(2, "0")}`;
```

Comment:
I recall that the color API was a bit "weird" the last time I played with it but Is there really no built-in way to generate the hex code?

Reply:
> I don't know how to turn own selections off.

With that, I guess you mean you do not know how to disable that in the color picker UI element? Well let's look at that in a next step.

Reply:
> but Is there really no built-in way to generate the hex code?

Yes, it's baffling but it can only parse hex. https://docs.gtk.org/gdk3/struct.RGBA.html
All top recommendations on stackoverflow are telling to do such string formatting.

> With that, I guess you mean you do not know how to disable that in the color picker UI element?

Yes, I'd like to turn that off because the API won't return any results for such unexpected colors.

Reply:
Ok, thanks for the clarification. :) Let's track the color selection problem [in a GitHub issue](https://github.com/ifl0w/RandomWallpaperGnome3/issues/168).